Assets/ModScripts/Submodules/WhosWho.cs

```diff
@@ -0,0 +1,293 @@
+            return 1;
+        }
+        else if (Module.Bomb.GetPortPlates().Any(x => x.Contains("DVI") && (x.Contains("StereoRCA") || x.Contains("PS2"))))
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] A DVI-D port is on the same port plate as a Stereo RCA port or a PS/2 port, using the sixth LED.", ModuleID);
+            return 5;
+        }
+        else if ((Module.Bomb.GetBatteryCount(1) > 0 && Module.Bomb.GetBatteryCount(2) == 0) || (Module.Bomb.GetBatteryCount(2) > 0 && Module.Bomb.GetBatteryCount(1) == 0))
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] There are only D batteries or AA batteries, using the fourth LED.", ModuleID);
+            return 3;
+        }
+        else
+        {
+            string[] ordinals = new string[]{"first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth"};
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] None of the conditions applied, using the {1} LED.", ModuleID, ordinals[Module.Bomb.GetSerialNumberNumbers().Sum() % 8]);
+            return Module.Bomb.GetSerialNumberNumbers().Sum() % 8;
+        }
+    }
+
+    private bool SameColorTwiceInARow(string[] colors)
+    {
+        for(int i = 0; i < colors.Length - 1; i++)
+            if (colors[i] == colors[i+1])
+                return true;
+        return false;
+    }
+
+    public override void OnBulbButtonPress(int Button)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.Bulbs[Button].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The {1} button was pressed when the component selection was [{2}] instead of [{3}].", ModuleID, (Button == 2) == Info.BulbInfo[4] ? "O" : "I", Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+        if ((Button == 2) == Info.BulbInfo[4])
+            O_PressTime.Start();
+        else
+            I_PressTime.Start();
+    }
+
+    public override void OnBulbButtonRelease(int Button)
+    {
+        O_PressTime.Stop();
+        I_PressTime.Stop();
+
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonRelease, Module.transform);
+
+        if (O_PressTime.Elapsed.TotalSeconds >= 1 || I_PressTime.Elapsed.TotalSeconds >= 1)
+        {
+            if (!submissionMode)
+            {
+                submissionMode = true;
+                UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Entering submission mode.", ModuleID);
+            }
+            else if (int.Parse(Module.WidgetText[2].text) == finalNumber)
```

Comment:
```suggestion
            else if (Info.NumberDisplay == finalNumber)
```
Don't use front-end stuff, get the module's info with Info instead.